### ISSUE TITLE
MQwen里重构的forward方法

### ISSUE THREAD

**Issue by kaseketsu** [WatchTower-Liu/VLM-learning#1]
`input_ids = torch.stack(new_input_ids, dim = 0).to(input_ids)`
这里的`to(input_ids)`有点不太明白，一般来说都是`to(device)`叭

**Comment by kaseketsu** [comment]
还有一个小问题
```
        elif input_ids is not None:
            input_shape = input_ids.size()
           ` input_ids = input_ids.view(-1, input_shape[-1]).contiguous()`
            batch_size = input_ids.shape[0]
        elif inputs_embeds is not None:
            input_shape = inputs_embeds.size()[:-1]
            batch_size = inputs_embeds.shape[0]
```
   一般来说, input_ids的size是(batch_size, seq_len)，这里的 ` batch_size = input_ids.shape[0]`      是不是放在  `input_ids = input_ids.view(-1, input_shape[-1]).contiguous()`前面更好捏，感觉这一段先合并前面的维度，再用`input_ids.shape[0]`获取batch_size有点奇怪，万一输入的input_ids是三维的，第一个维度是batch_size，合并前两个维度之后还能看做是原始输入的bs了么

**Comment by WatchTower-Liu** [comment]
@kaseketsu  to(tensor)  会实现包括device dtype在内的众多内容的转换， 获取batch_size是灵活的，这里是实验时一些调整导致的代码不规范。

**Comment by kaseketsu** [comment]
> @kaseketsu to(tensor) 会实现包括device dtype在内的众多内容的转换， 获取batch_size是灵活的，这里是实验时一些调整导致的代码不规范。

谢谢回复！作为一个多模态小白，您的代码让我学到很多！
